src/components/AccountLoginForm.jsx

```diff
@@ -6,107 +6,104 @@ import statefulForm from '../lib/statefulForm'
+    if (!renderers[type]) {
+      throw new Error('Unknown field type ' + type)
+    }
     const readOnly = name === 'login' && isUpdate
+
+    // Give focus only once
     const giveFocus = !alreadyFocused && !readOnly
     if (giveFocus) alreadyFocused = giveFocus
-    const inputName = `${name}_${connectorSlug}`
-    const description = field.hasDescription
-      ? <ReactMarkdownWrapper source={t(`connector.${connectorSlug}.description.field.${name}`)} />
-      : ''
-    const onEnterKey = () => {
-      if (((!isUpdate || hasEditableFields) && !isSuccess) && !submitting && submitEnabled) submit()
-    }
-    switch (field.type) {
-      case 'password':
-        return <div>
-          {description}
-          <PasswordField
-            label={t(`account.form.label.${name}`)}
-            name={inputName}
-            placeholder={t('account.form.placeholder.password')}
-            invalid={!!error}
-            giveFocus={giveFocus}
-            onEnterKey={onEnterKey}
-            noAutoFill
-            {...Object.assign({}, field, {
-              value: isUnloading ? '' : field.value
-            })}
-          />
-        </div>
-      case 'dropdown':
-        return <div>
-          {description}
-          <DropdownField label={t(`account.form.label.${name}`)} {...field} />
-        </div>
-      case 'folder':
-        return <div>
-          {description}
-          <FolderPickerField
-            readOnly={disableFolderPath}
-            label={t(`account.form.label.${name}`)}
-            {...field}
-          />
-        </div>
-      case 'checkbox':
-        // force boolean type here since it's just a checkbox
-        field.value = !!field.value
-        return <div>
-          {description}
-          <CheckboxField label={t(`account.form.label.${name}`)} {...field} />
-        </div>
-      default:
-        return <div>
-          {description}
-          <Field
-            label={t(`account.form.label.${name}`)}
-            name={inputName}
-            readOnly={readOnly}
-            invalid={!!error}
-            giveFocus={giveFocus && !readOnly}
-            onEnterKey={onEnterKey}
-            noAutoFill
-            {...Object.assign({}, field, {
-              value: isUnloading ? '' : field.value
-            })}
-          />
-        </div>
+
+    // Build common fields
+    const hydrate = hydrateFieldValue[name] || identity
+    const attributes = {
+      ...field, // TODO be more restrictive on what comes in
+      invalid: !!error,
+      onEnterKey,
+      giveFocus,
+      label: t(`account.form.label.${label || name}`),
+      value: isUnloading ? '' : hydrate(value)
     }
+    return <div>
+      {field.hasDescription &&
+        <ReactMarkdownWrapper source={t(`connector.${connectorSlug}.description.field.${name}`)} />}
+      {React.cloneElement(renderers[type](props), attributes)}
+    </div>
   }
 
   return (
     <div className={styles['account-form-login']}>
+      {/* Error */}
```

Comment:
Is this comment relevant ?

Reply:
mainly to be consistent with the two other comments